<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Yingxin Xu">

<title>README-code</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="README_files/libs/clipboard/clipboard.min.js"></script>
<script src="README_files/libs/quarto-html/quarto.js"></script>
<script src="README_files/libs/quarto-html/popper.min.js"></script>
<script src="README_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="README_files/libs/quarto-html/anchor.min.js"></script>
<link href="README_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="README_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="README_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="README_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="README_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">README-code</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Yingxin Xu </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="project-information" class="level1">
<h1>1. Project information</h1>
<section id="introduction-to-this-project" class="level2">
<h2 class="anchored" data-anchor-id="introduction-to-this-project">1.1 Introduction to this project</h2>
<p>This project is a collaboration with Dr.&nbsp;Karl Berator. The project aims to investigate the impact of a new treatment, activating factor 42, on the promotion of gene expression by growth factors.</p>
</section>
<section id="project-contacts" class="level2">
<h2 class="anchored" data-anchor-id="project-contacts">1.2 Project contacts</h2>
<p>For further collaboration on the project, please contact Dr.&nbsp;Karl Berator at karl.berator@gmail.com.</p>
<p>If you have any questions regarding the current analysis, please contact me (Yingxin Xu) at: Mobile: 0468862372; Email: a1871793@adelaide.edu.au</p>
</section>
<section id="project-software" class="level2">
<h2 class="anchored" data-anchor-id="project-software">1.3 Project software</h2>
<p>The language used for the project is R-4.4.0, and the analysis software is RStudio. If you need to install R language and RStudio, please follow the instructions on <a href="https://posit.co/download/rstudio-desktop/">this website</a> to download and install the version suitable for your device.</p>
</section>
<section id="how-to-download-the-project-to-your-device" class="level2">
<h2 class="anchored" data-anchor-id="how-to-download-the-project-to-your-device">1.4 How to download the project to your device</h2>
<p>This repository contains all the relevant files for the R project “Karl-Project”. Once you have installed R and RStudio, you only need to download this project to your local device, unzip it, and then open it in RStudio.</p>
</section>
<section id="introduction-to-project-folder-and-files" class="level2">
<h2 class="anchored" data-anchor-id="introduction-to-project-folder-and-files">1.5 Introduction to project folder and files</h2>
<section id="r-code" class="level3">
<h3 class="anchored" data-anchor-id="r-code">(1). <code>R-code</code>：</h3>
<p>This is a folder containing all the R code for the work previously done for Karl. All files are in Rmd (RMarkdown) format and can be directly executed in RStudio. The files are named following the format “year-month-day-code content.Rmd”.</p>
</section>
<section id="raw-data" class="level3">
<h3 class="anchored" data-anchor-id="raw-data">(2). <code>raw-data</code>:</h3>
<p>The folder contains the original data sent by Karl, which has not been altered and is labelled with the date of data reception.</p>
</section>
<section id="data" class="level3">
<h3 class="anchored" data-anchor-id="data">(3). <code>data</code>:</h3>
<p>The folder is used to store the modified data during the analysis process. Since the previous research did not involve storing the modified data, the folder currently only contains the original data.</p>
</section>
<section id="figs" class="level3">
<h3 class="anchored" data-anchor-id="figs">(4). <code>figs</code>:</h3>
<p>The folder contains figures generated during the analysis that may be needed for further use.</p>
</section>
<section id="tabs" class="level3">
<h3 class="anchored" data-anchor-id="tabs">(5). <code>tabs</code>:</h3>
<p>The folder contains tables generated during the analysis that may be needed for further use.</p>
</section>
<section id="slides" class="level3">
<h3 class="anchored" data-anchor-id="slides">(6). <code>slides</code>:</h3>
<p>The folder contains a presentation on Exploratory Data Analysis (EDA) provided to Karl.</p>
</section>
<section id="font" class="level3">
<h3 class="anchored" data-anchor-id="font">(7). <code>font</code>:</h3>
<p>This folder contains the a font file required for plotting.</p>
</section>
<section id="report" class="level3">
<h3 class="anchored" data-anchor-id="report">(8). <code>report</code>:</h3>
<p>The folder stores the reports generated during the analysis.</p>
</section>
</section>
</section>
<section id="current-analysis" class="level1">
<h1>2. Current analysis</h1>
<section id="introduction-to-data-set" class="level2">
<h2 class="anchored" data-anchor-id="introduction-to-data-set">2.1 Introduction to data set</h2>
<p>The data set <code>2024-03-04-WIF-tis4d.xlsx</code> is stored in the folder <code>raw-data</code>. It contains following terms:</p>
<ol type="1">
<li><p><code>cell_line</code>: The two cell line types used for the experiment, Wild-type and Cell-type 101, character.</p></li>
<li><p><code>Treatment</code>: This variable describes whether to add new treatment. Added: Activating factor 42; Not added: Placebo (saline added instead), character.</p></li>
<li><p><code>name</code>: Names of different cell lines. There are two cell lines for each identical cell line type and the same treatment. There were 8 cell lines in total, character.</p></li>
<li><p><code>conc</code>: Different concentration of a growth factor, from <span class="math inline">\(0\)</span> to <span class="math inline">\(10\)</span>, numeric.</p></li>
<li><p><code>gene_expression</code>: Gene expression level, numeric.</p></li>
</ol>
</section>
<section id="eda-2024-3-21" class="level2">
<h2 class="anchored" data-anchor-id="eda-2024-3-21">2.2 EDA (2024-3-21)</h2>
<p>Karl requested an EDA on this data set and a presentation be prepared to explain the results to his boss. The complete code for performing EDA on the data is located in the <code>R-code</code> folder, named <code>2024-03-21-EDA.Rmd</code>. The slides are located in the <code>slide</code> folder, named <code>2024-03-21-EDA.pptx</code>.</p>
</section>
<section id="sec-figure" class="level2">
<h2 class="anchored" data-anchor-id="sec-figure">2.3 Reproducing an EDA figure (2024-04-01)</h2>
<p>Karl requested to plot the following figure and save as an .tiff file (9in x 6in) with a resolution of 500, and use Time New Roman font.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-01" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="README_files/figure-html/fig-01-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Gene expression vs.&nbsp;growth factor concentrations plot. This plot is divided into two parts based on cell line type, labelled as A and B, and different colors are used to represent different treatments. The tags on the graph represent different cell line individuals.</figcaption>
</figure>
</div>
</div>
</div>
<p>The complete code for this is located in the <code>R-code</code> folder, named <code>2024-04-01-Figure.Rmd</code>. The figure was saved in the <code>figs</code> folder, named <code>gene_fig.tiff</code>.</p>
</section>
<section id="calculate-sample-size-for-grant-2024-05-22" class="level2">
<h2 class="anchored" data-anchor-id="calculate-sample-size-for-grant-2024-05-22">2.4 Calculate sample size for grant (2024-05-22)</h2>
<p>Karl requested to calculate sample size for a grant application. They were going to do a linear regression of gene expression with five predictors: concentration, cell age, treatment (two levels), cell type (two levels), and media (two levels). They have done a previous study and got an <span class="math inline">\(R^2\)</span> of 0.1 between the predictors and the response level. They wanted a power of <span class="math inline">\(90\%\)</span> and a significance level of <span class="math inline">\(0.05\)</span>.</p>
<p>The calculation shows that they need a total of <span class="math inline">\(154\)</span> samples.</p>
<p>The complete code for this sample size calculation is located in the <code>R-code</code> folder, named <code>2024-05-22-sample size.Rmd</code>. For better understanding, we have also generated a report named <code>2024-05-22-sample-size.pdf</code>, located in the <code>report</code> folder.</p>
</section>
<section id="fit-predictive-models-2024-05-28" class="level2">
<h2 class="anchored" data-anchor-id="fit-predictive-models-2024-05-28">2.5 Fit predictive models (2024-05-28)</h2>
<p>The final step of the current analysis is to fit a predictive model to the gene expression data.</p>
<p>As mentioned earlier, the purpose of this project is to explore the impact of different treatments on the promotion effect of growth factors on gene expression. Among the five variables, gene expression is the response variable, while the other four are independent variables, consisting of three categorical variables and one numerical variable.</p>
<p>From the previous EDA, differences in gene expression across various levels of cell line type and treatment were observed. Generally, gene expression tends to increase with higher concentrations of growth factors. As depicted in Figure <a href="#fig-01">Figure&nbsp;1</a> in the <a href="#sec-figure">Section&nbsp;2.3</a>, even with identical treatment and growth factor concentrations, disparities in gene expression persist among different individuals of the same cell line type. Consequently, it can be considered to include the variable <code>name</code> as either an independent variable or a random effect (since in future predictions, we might encounter new cell line individuals, so fitting parameters only for the current eight cell line individuals would be unreasonable).</p>
<p>Therefore, two models were considered: a linear model and a linear mixed-effects model. In the linear model, <code>name</code> is treated as a regular categorical variable. In the mixed-effects model, <code>name</code> is treated as a random effect. Additionally, since the study aims to investigate the effect of treatment on the relationship between growth factor and gene expression, we also tested the necessity of adding the interaction term between <code>Treatment</code> and <code>conc</code> in both the linear model and the linear mixed-effects model. The results indicate that adding this interaction term is crucial for both models. So the following two models were compared:</p>
<ol type="1">
<li><p>Linear model:</p>
<p>gene_expression ~ Treatment * conc + cell_line + name</p></li>
<li><p>Linear mixed-effects model (1|name indicates that <code>name</code> is a random factor):</p>
<p>gene_expression ~ Treatment * conc + cell_line + (1|name)</p></li>
</ol>
<p>Subsequently, the two models were compared using AIC, BIC, and regression diagnostic plots. The comparison results indicated that the linear model slightly outperformed the linear mixed-effects model. However, the difference between the two models was very small. Considering that new cell line individuals will be encountered in future predictions using the model, <code>name</code> should still be treated as a random effect. Therefore, the final model chosen is a linear mixed-effects model with <code>gene_expression</code> as the response variable, <code>Treatment</code>, <code>conc</code>, and <code>cell_line</code> as fixed effects, <code>name</code> as a random effect, and an interaction term between <code>Treatment</code> and <code>conc</code>.</p>
<p>The complete model fitting process code can be found in the <code>R-code</code> folder in the <code>2024-05-28-Fit models.Rmd</code> file. For a more detailed model analysis process, please refer to the <code>2024-05-29-IMRaD-Report.pdf</code> file in the <code>report</code> folder.</p>
</section>
<section id="imrad-report-2024-05-29" class="level2">
<h2 class="anchored" data-anchor-id="imrad-report-2024-05-29">2.6 IMRaD report (2024-05-29)</h2>
<p>Finally, Karl requested an IMRaD report that encompasses all the key points from the analysis process described above. This report serves as a summary of the previous work but does not include all the code. The R code to generate this IMRaD report can be found in the <code>R-code</code> folder, named <code>2024-05-29-IMRaD-Report.Rmd</code>. The corresponding PDF report is in the <code>report</code> folder, named <code>2024-05-29-IMRaD-Report.pdf</code>.</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>